docs/products/kafka/howto/flink-with-aiven-for-kafka.rst

```diff
@@ -0,0 +1,169 @@
+Where Apache Kafka® excels at receiving and sending event streams, Flink consumes, transforms, aggregates, and enriches your data.
+
+The example in this article shows you how to create a simple Java Flink job that reads data from a Kafka topic, processes it, and then pushes it to a different Kafka topic.
+
+.. Note::
+
+   If you want to experience the power of streaming SQL transformations with Flink, Aiven provides a hosted :doc:`/docs/products/flink` with built-in data flow integration with Aiven for Apache Kafka®.
+
+   This example demonstrates using the Java API on a `local installation of Apache Flink 1.15.1 <https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/try-flink/local_installation/>`_, but it can be applied to use Aiven for Apache Kafka with any self-hosted cluster.
+
+.. _kafka-flink-prereq:
+
+Prerequisites
+-------------
+
+You need an Aiven for Apache Kafka service up and running with two topics, named ``test-flink-input`` and ``test-flink-output`` already created. Furthermore, for the example, you need to collect the following information about the Aiven for Apache Kafka service:
+
+* ``APACHE_KAFKA_HOST``: The hostname of the Apache Kafka service
+* ``APACHE_KAFKA_PORT``: The port of the Apache Kafka service
+
+You need to have `Apache Maven™ <https://maven.apache.org/install.html>`_ installed to build the example.
+
+Setup the truststore and keystore
+''''''''''''''''''''''''''''''''''
+
+Create a :doc:`Java keystore and truststore <keystore-truststore>` for the Aiven for Apache Kafka service.
+For the following example we assume:
+
+* The keystore is available at ``KEYSTORE_PATH/client.keystore.p12``
+* The truststore is available at ``TRUSTSTORE_PATH/client.truststore.jks``
+* For simplicity, the same secret (password) is used for both the keystore and the truststore, and is shown here as ``KEY_TRUST_SECRET``
+
+
+Use Flink with Aiven for Apache Kafka
+-------------------------------------
+
+The following example shows how to customise the ``DataStreamJob`` generated from the `Quickstart <https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/configuration/overview/>`_ to work with Aiven for Apache Kafka.
+
+1. Generate a Flink job skeleton named ``capitalizer`` using the Maven archetype:
+
+   .. code:: shell
+
+      mvn archetype:generate -DinteractiveMode=false \
+        -DarchetypeGroupId=org.apache.flink -DarchetypeArtifactId=flink-quickstart-java \
+        -DarchetypeVersion=1.15.1 -DgroupId=com.example -DartifactId=capitalizer \
+        -Dversion=0.0.1-SNAPSHOT
+
+2. Uncomment the Kafka connector in `pom.xml`:
+
+   .. code:: xml
+
+      <dependency>
+        <groupId>org.apache.flink</groupId>
+        <artifactId>flink-connector-kafka</artifactId>
+        <version>${flink.version}</version>
+      </dependency>
+
+Customizing the ``DataStreamJob`` application
```

Comment:
Expanding on what @snuyanzin commented of full example. There are two alternatives more, diff file to load and patch on top of generated example. Or a quickstart script approach what Flink documentation uses on https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/configuration/overview/#getting-started. Hard to navigate, but click the `Gradle` tab and scroll down a bit to `Quickstart script`.